tests/tests_ut/az_ulib_ipc_ut/az_ulib_ipc_ut.c

```diff
@@ -1131,156 +1772,217 @@ static void az_ulib_ipc_unpublish_with_valid_interface_instance_succeed(void** s
+          AZ_SPAN_FROM_STR(MY_INTERFACE_1_NAME),
+          MY_INTERFACE_123_VERSION,
+          &dafault_package_version),
       AZ_OK);
-  g_count_acquire = 0;
-
-  /// act
-  az_result result = az_ulib_ipc_try_get_interface(
-      AZ_SPAN_FROM_STR(MY_INTERFACE_1_2_INTERFACE_NAME),
-      MY_INTERFACE_1_2_INTERFACE_VERSION,
-      AZ_ULIB_VERSION_GREATER_THAN,
-      &greater_interface_handle);
 
   /// assert
-  assert_int_equal(result, AZ_OK);
-  assert_ptr_not_equal(interface_handle, greater_interface_handle);
   assert_int_equal(g_lock_diff, 0);
-  assert_int_equal(g_count_acquire, 1);
+  assert_int_equal(g_count_acquire, 3);
+  assert_int_equal(package_1, dafault_package_version);
+  assert_int_not_equal(package_1, package_2);
 
   /// cleanup
-  assert_int_equal(az_ulib_ipc_release_interface(interface_handle), AZ_OK);
-  assert_int_equal(az_ulib_ipc_release_interface(greater_interface_handle), AZ_OK);
+  assert_int_equal(az_ulib_ipc_release_interface(package_1), AZ_OK);
+  assert_int_equal(az_ulib_ipc_release_interface(package_2), AZ_OK);
+  assert_int_equal(az_ulib_ipc_release_interface(dafault_package_version), AZ_OK);
   unpublish_interfaces_and_deinit_ipc();
 }
 
-static void az_ulib_ipc_try_get_interface_version_lower_than_succeed(void** state)
+static void az_ulib_ipc_try_get_interface_default_anbiguous_name_failed(void** state)
 {
   /// arrange
   (void)state;
-  az_ulib_ipc_interface_handle interface_handle;
-  az_ulib_ipc_interface_handle lower_interface_handle;
+  az_ulib_ipc_interface_handle dafault_package_version;
   init_ipc_and_publish_interfaces();
-  assert_int_equal(
-      az_ulib_ipc_try_get_interface(
-          AZ_SPAN_FROM_STR(MY_INTERFACE_1_123_INTERFACE_NAME),
-          MY_INTERFACE_1_123_INTERFACE_VERSION,
-          AZ_ULIB_VERSION_EQUALS_TO,
-          &interface_handle),
-      AZ_OK);
-  g_count_acquire = 0;
 
   /// act
   az_result result = az_ulib_ipc_try_get_interface(
-      AZ_SPAN_FROM_STR(MY_INTERFACE_1_123_INTERFACE_NAME),
-      MY_INTERFACE_1_123_INTERFACE_VERSION,
-      AZ_ULIB_VERSION_LOWER_THAN,
-      &lower_interface_handle);
+      AZ_SPAN_EMPTY,
+      AZ_SPAN_EMPTY,
+      AZ_ULIB_VERSION_DEFAULT,
+      AZ_SPAN_FROM_STR(MY_INTERFACE_1_NAME),
+      MY_INTERFACE_123_VERSION,
+      &dafault_package_version);
 
   /// assert
-  assert_int_equal(result, AZ_OK);
-  assert_ptr_not_equal(interface_handle, lower_interface_handle);
   assert_int_equal(g_lock_diff, 0);
   assert_int_equal(g_count_acquire, 1);
+  assert_int_equal(result, AZ_ERROR_ULIB_AMBIGUOUS);
 
   /// cleanup
-  assert_int_equal(az_ulib_ipc_release_interface(interface_handle), AZ_OK);
-  assert_int_equal(az_ulib_ipc_release_interface(lower_interface_handle), AZ_OK);
   unpublish_interfaces_and_deinit_ipc();
 }
 
-static void az_ulib_ipc_try_get_interface_version_lower_or_equal_succeed(void** state)
+/* If the device is not AZ_SPAN_EMPTY, the az_ulib_ipc_try_get_interface shall return
+ * AZ_ERROR_NOT_IMPLEMENTED. */
```

Comment:
This is something that will be supported in future versions, correct? If so, can you make a note of it?

Reply:
Sure.